### ISSUE TITLE
Can I change the graphics card number if I want to continue training?

### ISSUE THREAD

**Issue by pppkkk611** [ultralytics/ultralytics#5918]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When continuing training after Yolo interruption, can I change the graphics card number if I want to continue training? I used device=0, 1 now I want to use device=2, 3
Yolo train resume model=/home/yolov8/runs/detect/train32/weights/last. pt device=2,3
Not effective

**Comment by pppkkk611** [comment]
The official documents do not seem to give me the situation I encountered

**Comment by pppkkk611** [comment]
The official documents do not seem to give me the situation I encountered

**Comment by glenn-jocher** [comment]
@pppkkk611 thanks for bringing up your concern. In order to effectively address your issue, could you please provide more specific details about the problem you encountered?

When it comes to switching the GPU device in your training configuration for YOLOv8, yes, you can change device numbers as per availability and requirement. If device=0, 1 were used previously and now you want to use device=2, 3, that should theoretically be possible.

However, if you're experiencing difficulties with this, there could be multiple causes. For example, it could be due to GPU usage by other processes, CUDA-related issues, or even insufficient GPU memory.

So, if you could provide more information about the error messages you're receiving, or the behavior that's occurring when you try to switch devices, it would be much easier to provide a more accurate and effective solution.

As always, please ensure you're using the latest version of the YOLOv8 codebase, and that all your software dependencies (such as CUDA, cuDNN, PyTorch etc.) are up to date and correctly installed.

**Comment by pppkkk611** [comment]
I use commands：yolo train resume model=/home/yolov8/runs/detect/train32/weights/last.pt device=2,3 
But the output information still displays：Ultralytics YOLOv8.0.183 🚀 Python-3.8.0 torch-2.0.1+cu117 
CUDA:0 (NVIDIA GeForce RTX 3090, 24260MiB)
CUDA:1 (NVIDIA GeForce RTX 3090, 24260MiB)
engine/trainer: task=detect, mode=train, model=/home/yolov8/runs/detect/train32/weights/last.pt, data=/home/yolov8/datasets/coco.yaml, epochs=150, patience=50, batch=128, imgsz=640, save=True, save_period=-1, cache=False, **device=(0, 1)**, workers=8, project=None, name=None, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, stream_buffer=False, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.0, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, tracker=botsort.yaml, save_dir=runs/detect/train32
DDP: debug command /home/.conda/envs/yolov8/bin/python -m torch.distributed.run --nproc_per_node 2 --master_port 59247 /home/.config/Ultralytics/DDP/_temp_m4quqg_4140183465151648.py

**Comment by glenn-jocher** [comment]
@pppkkk611 this seems like a potential issue with specifying the device for training in the command line. From the output you've shared it's clear that the devices being used for CUDA are still 0 and 1, when you want them to be 2 and 3.

While you're resuming the training, make sure that the appropriate GPU devices are available and not being used by other processes. Also verify the IDs of the GPUs in your system using the `nvidia-smi` command.

Additionally, please ensure that the command you're writing to resume the model is correct. It seems you've put "model=" before specifying the model, which could potentially lead to a problem. 

If everything looks good, it might be a bug. In which case, try to update YOLOv8 to the latest version to see if the problem has been addressed in a newer release.

Let me know if this helps, or if you're still encountering problems!

**Comment by yellowjs0304** [comment]
@glenn-jocher @pppkkk611 Have you figured out this issue? 
I have same issue.
It seems that the GPU is set up with hyperparameters stored in the model.

**Comment by pppkkk611** [comment]
> 你弄清楚这个问题了吗？我有同样的问题。 GPU 似乎是使用存储在模型中的超参数设置的。

No, I haven't solved it. I've tried many ways. Finally, due to the need to modify the model and retrain it.

**Comment by pppkkk611** [comment]
> @glenn-jocher @pppkkk611 Have you figured out this issue? I have same issue. It seems that the GPU is set up with hyperparameters stored in the model.

No, I haven't solved it. I've tried many ways. Finally, due to the need to modify the model and retrain it.

**Comment by glenn-jocher** [comment]
@yellowjs0304 I'm sorry to hear you're still experiencing this issue. It does seem like the GPU device is being determined from the original hyperparameters saved in the model and not from the command line input when resuming training.

One possibility could be to alter the hyperparameters saved in the model to specify your desired GPU device. However, this would involve digging into the model's saved state and could be complex.

As potential workarounds, you might consider pausing other processes on your original GPUs to free them up for training, or else starting your training from scratch on GPUs 2 and 3.

I understand these solutions aren't ideal and it's a limitation in the YOLOv8's current design. I appreciate your patience in this matter. Your feedback is important to us and informs our continuous efforts to improve YOLOv8's performance and flexibility.
